ironfist/src/cpp/combat/combat.cpp

```diff
@@ -486,7 +486,6 @@ void army::SpecialAttack() {
 	if (!gpCombatManager->heroes[this->owningSide] || !gpCombatManager->heroes[this->owningSide]->HasArtifact(ARTIFACT_AMMO_CART))
 		--this->creature.shots;
 
-	char monAttack = this->creature.attack;
```

Comment:
I verified that this is safe to remove. It was not obvious because any of the function calls in between could have modified this field.